### PR TITLE
05-JS Functions

### PR DESCRIPTION
**Pull request by aaronpowell**
Here's the first draft of the JS Azure Functions post for the 5th of September

### CODE REVIEW

File: website/blog/2022-09-05/index.md [hunk 1]
```diff
@@ -33,31 +33,152 @@ tags: [serverless-september, 30-days-of-serverless, serverless-hacks, zero-to-he
...
+The application we're going to be building today will use the GitHub API to return a random commit message, so that we don't need to come up with one ourselves! After all, naming things can be really hard! 🤣 
 
-## Section 2
+Ensure you have Node.js 18 installed, a text editor (I'll use VS Code in this post), and a terminal, then we're ready to go.
 
-## Section 3
+## Creating an Azure Function
 
-## Section 4
+To create our Azure Function, we're going to use the [Azure Functions CLI](), which we can install using npm:
```
Comment (nitya) [inline website/blog/2022-09-05/index.md:59]: Missing link for Azure Functions CLI

File: website/blog/2022-09-05/index.md [hunk 2]
```diff
@@ -33,31 +33,152 @@ tags: [serverless-september, 30-days-of-serverless, serverless-hacks, zero-to-he
...
+
+The you'll get some commit messages:
+
+![A series of commit messages from the GitHub Search API](./img/04.png)
+
+🎉 There we go, we've created an Azure Function which is used as a proxy to another API, that we call using native `fetch` in Node.js 18, and return a subset of the JSON payload.
+
+## Exercise - Support searching
+
+The GitHub Serach API allows you to provide search parameters via the `q` query string. In this sample, we hard-coded it to be `language:javascript`, but as a follow on exercise, expand the Function to allow the caller to provide the search terms as a query string to the Azure Function, which is passed to the GitHub API. Hint - have a look at the `req` argument.
```
Comment (nitya) [inline website/blog/2022-09-05/index.md:178]: Minor type - should read "Search API"

**Review by nitya — CHANGES_REQUESTED** [review]
Left a few comments - also wondering if you could grab screenshots in light mode (for readability) if possible. If not, we can work with this. If you finalize and submit PR, I can prep the metadata and banner and have this ready to go.

File: website/blog/2022-09-05/index.md (hunk 1 above)
Comment (aaronpowell) [inline website/blog/2022-09-05/index.md:59]:
```suggestion
To create our Azure Function, we're going to use the [Azure Functions CLI](https://docs.microsoft.com/azure/azure-functions/functions-run-local?tabs=v4%2Cwindows%2Ccsharp%2Cportal%2Cbash), which we can install using npm:
```

File: website/blog/2022-09-05/index.md (hunk 2 above)
Comment (aaronpowell) [inline website/blog/2022-09-05/index.md:178]:
```suggestion
The GitHub Serach API allows you to provide search parameters via the `q` query string. In this sample, we hard-coded it to be `language:javascript`, but as a follow-on exercise, expand the Function to allow the caller to provide the search terms as a query string to the Azure Function, which is passed to the GitHub Search API. Hint - have a look at the `req` argument.
```

Comment (aaronpowell) [inline website/blog/2022-09-05/index.md:178]:
```suggestion
The GitHub Serach API allows you to provide search parameters via the `q` query string. In this sample, we hard-coded it to be `language:javascript`, but as a follow-on exercise, expand the Function to allow the caller to provide the search terms as a query string to the Azure Function, which is passed to the GitHub Search API. Hint - have a look at the `req` argument.
```

**Comment by aaronpowell** [comment]
I've left the dark theme pictures in there, but changed to light theme for default.

I wonder if we could do something that would switch between them when you change theme

**Review by Kingris9090 — APPROVED** [review]

**Review by Kingris9090 — APPROVED** [review]

**Review by nitya — APPROVED** [review]
Approved  - yes, there is a way to pick theme-specific image for post.

**Pull request merged**